### PR TITLE
Add Moxy and Conductor

### PR DESCRIPTION
**Pull request by rougsig**
Add Conductor
Add Moxy

Add Conductor+Moxy sample

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,8 +15,23 @@ Mobile app to rule all Flipper's family
...
+- `app` - Main application module with UI
+- `core` - Core library with deps and utils
+- `bridge` - Communication between android and Flipper
+- `modules` - Community widget module
 
+## Layer arch
+
+```
+├── app
+│   ├── feature1
```
Comment (LionZXY) [inline README.md:27]: А давай наоборот. Слои, а внутри папки

File: core/src/main/java/com/flipper/core/view/BaseController.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.viewbinding.ViewBinding
+import com.bluelinelabs.conductor.Controller
+import moxy.MvpDelegate
+
+abstract class BaseController<VB : ViewBinding> : Controller {
+  // Lazy used for prevent leaking `this`
+  private val mvpDelegate by lazy(LazyThreadSafetyMode.NONE) { MvpDelegate(this) }
```
Comment (LionZXY) [inline core/src/main/java/com/flipper/core/view/BaseController.kt:13]: May be call it `presenter`?

**Review by LionZXY — CHANGES_REQUESTED** [review]

File: app/build.gradle.kts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 plugins {
   id("com.android.application")
   id("kotlin-android")
+  id("kotlin-kapt")
   id("kotlin-android-extensions")
```
Comment (MaximPestryakov) [inline app/build.gradle.kts:5]: I guess it should be removed if we use `ViewBinding`

File: core/src/main/java/com/flipper/core/view/BaseController.kt (hunk 2 above)
Comment (rougsig) [reply core/src/main/java/com/flipper/core/view/BaseController.kt:13]: No we can't. It's delegate for provide moxy livecycle.

File: README.md (hunk 1 above)
Comment (rougsig) [reply README.md:27]:
```
├── app
│   ├── feature1
│   │   ├── data
│   │   ├── view
├──   interactor
│   ├── feature1
├──   repository
│   ├── feature1
```

Maybe that?

File: core/src/main/java/com/flipper/core/view/ViewInflater.kt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package com.flipper.core.view
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+
+typealias ViewInflater<T> = (LayoutInflater, ViewGroup, Boolean) -> T
```
Comment (MaximPestryakov) [inline core/src/main/java/com/flipper/core/view/ViewInflater.kt:6]: It is better to explicitly name the parameters like `ViewGroup` and `Boolean`:
```kotlin
typealias ViewInflater<T> = (LayoutInflater, parent: ViewGroup, attachToParent: Boolean) -> T
```

**Review by MaximPestryakov — APPROVED** [review]
Ok, but I prefer MVI instead of MVP

File: app/build.gradle.kts (hunk 3 above)
Comment (rougsig) [reply app/build.gradle.kts:5]: Yes we can, but `kotlin-android-extensions` also can be used for `@Parcelize`. But for now i will remove that plugin from modules.

Comment (rougsig) [reply app/build.gradle.kts:5]: Fixed

File: core/src/main/java/com/flipper/core/view/ViewInflater.kt (hunk 4 above)
Comment (rougsig) [reply core/src/main/java/com/flipper/core/view/ViewInflater.kt:6]: Fixed

**Review by LionZXY — APPROVED** [review]

**Pull request merged**